### ISSUE TITLE
numpy 1.11.0 nonuniform binning

### ISSUE THREAD

**Issue by bla1089** [pandas-dev/pandas#14627]
`numpy 1.11.0` allows for several strings as inputs to create non-uniform binning. It would be useful if this could be implemented in pandas.

[See the `numpy.histogram` docs.](https://docs.scipy.org/doc/numpy/reference/generated/numpy.histogram.html)

**Comment by jreback** [comment]
can you update to show an example usage

**Comment by bla1089** [comment]
As of right now, the `numpy` code is in `numpy/lib/function_base.py`.

**Comment by bla1089** [comment]
There's an example over at [`astropy`](http://docs.astropy.org/en/stable/visualization/histogram.html)using `freedman`, `scott`, and several not in `pandas`.

**Comment by TomAugspurger** [comment]
@bla1089 your request is for plotting specifically, correct?

**Comment by jreback** [comment]
we won't actually use the numpy code, rather just call it to create the bins

**Comment by bla1089** [comment]
@TomAugspurger nope. The `astropy` example is just a good way to illustrate the result.

**Comment by TomAugspurger** [comment]
It'd be helpful if you updated the original post with a full example of what your looking for, and why you think it'd be useful.

**Comment by jreback** [comment]
no objection to dispatching on a string to bins
if the numpy version is acceptable 

a PR would be great

**Comment by bla1089** [comment]
What would I need to do in the way of testing?

---------------------
B. L. Alterman
Candidate, Applied Physics
Solar and Heliospheric Research Group
Climate and Space Sciences and Engineering
University of Michigan
balterma@umich.edu

**Comment by jreback** [comment]
follow along in test_tile.py how we test the bins argument

**Comment by bla1089** [comment]
Do you know of a simple way to calculate the precision at which we would need to store the `np. histogram_bin_edges` to maintain unique edges?

**Comment by jreback** [comment]
not really sure what you mean

**Comment by bla1089** [comment]
From what I can tell, `pd.cut` creates `Categorical`s with left and right edges that are stored to `precision`, which defaults to 3 decimal places. What if binning routine returns edges that are duplicate at that precision and we need a precision of 4 or 5 decimal places to ensure each edge is unique? How can we calculate that precision, the minimal precision at which the bin edges are unique?

**Comment by jreback** [comment]
and your example of this is ?

**Comment by TomAugspurger** [comment]
It should be there.
http://pandas-docs.github.io/pandas-docs-travis/contributing.html#creating-a-python-environment
(though another PR is adjusting this right now)

**Comment by mroeschke** [comment]
Appears there hasn't been much activity or community support for this feature in a while so closing. Happy to reopen if there's renewed support
